packages/x-charts/src/hooks/animation/useAnimate.ts

```diff
@@ -0,0 +1,65 @@
+import useForkRef from '@mui/utils/useForkRef';
+import * as React from 'react';
+import { useAnimateInternal } from '../../internals/animation/useAnimateInternal';
+
+export interface UseAnimateParams<Props extends {}, Elem extends Element, T extends {} = Props> {
+  createInterpolator: (lastProps: Props, newProps: Props) => (t: number) => Props;
+  transformProps: (props: Props) => T;
+  applyProps: (element: Elem, props: T) => void;
+  skip?: boolean;
+  initialProps?: Props;
+  ref?: React.Ref<Elem>;
+}
```

Comment:
Mayb here is the good place to add JSDocs

Reply:
Done 👍 let me know if what I wrote makes sense